File: app/src/main/java/com/example/doan/Adapters/StoreAdapter.java
package com.example.doan.Adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.doan.Models.Store;
import com.example.doan.R;
import com.google.android.material.button.MaterialButton;
import com.google.android.material.chip.Chip;

import java.util.List;

public class StoreAdapter extends RecyclerView.Adapter<StoreAdapter.StoreViewHolder> {

    private Context context;
    private List<Store> stores;

    public StoreAdapter(Context context, List<Store> stores) {
        this.context = context;
        this.stores = stores;
    }

    @NonNull
    @Override
    public StoreViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_store, parent, false);
        return new StoreViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull StoreViewHolder holder, int position) {
        Store store = stores.get(position);
        holder.bind(store);
    }

    @Override
    public int getItemCount() {
        return stores.size();
    }

    public void updateStores(List<Store> newStores) {
        this.stores = newStores;
        notifyDataSetChanged();
    }

    class StoreViewHolder extends RecyclerView.ViewHolder {
        TextView tvStoreName, tvStoreAddress, tvStorePhone, tvStoreHours;
        Chip chipStoreStatus;
        MaterialButton btnEditStore, btnDeleteStore;

        public StoreViewHolder(@NonNull View itemView) {
            super(itemView);
            tvStoreName = itemView.findViewById(R.id.tv_store_name);
            tvStoreAddress = itemView.findViewById(R.id.tv_store_address);
            tvStorePhone = itemView.findViewById(R.id.tv_store_phone);
            tvStoreHours = itemView.findViewById(R.id.tv_store_hours);
            chipStoreStatus = itemView.findViewById(R.id.chip_store_status);
            btnEditStore = itemView.findViewById(R.id.btn_edit_store);
            btnDeleteStore = itemView.findViewById(R.id.btn_delete_store);
        }

        public void bind(Store store) {
            tvStoreName.setText(store.getName());
            tvStoreAddress.setText(store.getAddress());
            tvStorePhone.setText(store.getPhone() != null ? store.getPhone() : "N/A");
            
            // Mock hours - replace with actual data if available
            tvStoreHours.setText("8:00 - 22:00");
            
            // Status
            chipStoreStatus.setText("Hoạt động");
            chipStoreStatus.setChipBackgroundColorResource(R.color.success);

            // Edit button
            btnEditStore.setOnClickListener(v -> {
                Toast.makeText(context, "Sửa: " + store.getName(), Toast.LENGTH_SHORT).show();
            });

            // Delete button
            btnDeleteStore.setOnClickListener(v -> {
                Toast.makeText(context, "Xóa: " + store.getName(), Toast.LENGTH_SHORT).show();
            });
        }
    }
}

File: app/src/main/java/com/example/doan/Adapters/StoreAdapter.kt
package com.example.doan.Adapters

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.doan.Models.Store
import com.example.doan.R
import com.google.android.material.button.MaterialButton
import com.google.android.material.chip.Chip

class StoreAdapter(
    private val context: Context,
    private var stores: List<Store>,
    private val onStoreClick: (Store) -> Unit // Click listener
) : RecyclerView.Adapter<StoreAdapter.StoreViewHolder>() {

    private var distanceMap: Map<Int, String> = emptyMap()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StoreViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_store, parent, false)
        return StoreViewHolder(view)
    }

    override fun onBindViewHolder(holder: StoreViewHolder, position: Int) {
        val store = stores[position]
        val distanceInfo = distanceMap[store.id]
        holder.bind(store, distanceInfo, position == 0 && distanceMap.isNotEmpty())
    }

    override fun getItemCount(): Int = stores.size

    fun updateStores(newStores: List<Store>) {
        this.stores = newStores
        notifyDataSetChanged()
    }
    

    fun updateStoresWithDistance(newStores: List<Store>, distances: Map<Int, String>) {
        this.stores = newStores
        this.distanceMap = distances
        notifyDataSetChanged()
    }

    inner class StoreViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val tvStoreName: TextView = itemView.findViewById(R.id.tv_store_name)
        private val tvStoreAddress: TextView = itemView.findViewById(R.id.tv_store_address)
        private val tvStorePhone: TextView = itemView.findViewById(R.id.tv_store_phone)
        private val tvStoreHours: TextView = itemView.findViewById(R.id.tv_store_hours)
        private val chipStoreStatus: Chip = itemView.findViewById(R.id.chip_store_status)
        private val tvDistance: TextView? = itemView.findViewById(R.id.tv_store_distance)
        private val btnViewOnGoogleMaps: MaterialButton = itemView.findViewById(R.id.btn_view_on_google_maps)

        fun bind(store: Store, distanceInfo: String?, isNearest: Boolean) {
            tvStoreName.text = store.storeName
            tvStoreAddress.text = store.address
            tvStorePhone.text = store.phone ?: "N/A"
            
            // Hiển thị giờ mở cửa
            val openTime = store.openTime ?: "8:00"
            val closeTime = store.closeTime ?: "22:00"
            tvStoreHours.text = "$openTime - $closeTime"
            
            // Hiển thị khoảng cách nếu có
            tvDistance?.let { tv ->
                if (distanceInfo != null) {
                    tv.visibility = View.VISIBLE
                    tv.text = "📍 $distanceInfo"
                } else {
                    tv.visibility = View.GONE
                }
            }
            
            // Status - highlight quán gần nhất
            if (isNearest) {
                chipStoreStatus.text = "Gần nhất ⭐"
                chipStoreStatus.setChipBackgroundColorResource(R.color.primary)
            } else {
                chipStoreStatus.text = "Hoạt động"
                chipStoreStatus.setChipBackgroundColorResource(R.color.success)
            }
            
            // Xử lý nút xem trên Google Maps
            btnViewOnGoogleMaps.setOnClickListener {
                openGoogleMaps(store)
            }
            
            itemView.setOnClickListener {
                onStoreClick(store)
            }
        }
        
        private fun openGoogleMaps(store: Store) {
            // Ưu tiên sử dụng tọa độ nếu có
            val lat = store.latitude
            val lng = store.longitude
            val storeName = store.storeName ?: "Cửa hàng"
            val address = store.address ?: ""
            
            val uri = if (lat != 0.0 && lng != 0.0) {
                // Mở Google Maps với tọa độ chính xác
                Uri.parse("https://www.google.com/maps/search/?api=1&query=$lat,$lng")
            } else if (address.isNotEmpty()) {
                // Fallback: tìm kiếm theo địa chỉ
                val encodedAddress = Uri.encode("$storeName, $address")
                Uri.parse("https://www.google.com/maps/search/?api=1&query=$encodedAddress")
            } else {
                // Fallback cuối: tìm theo tên
                val encodedName = Uri.encode(storeName)
                Uri.parse("https://www.google.com/maps/search/?api=1&query=$encodedName")
            }
            
            val intent = Intent(Intent.ACTION_VIEW, uri)
            context.startActivity(intent)
        }
    }
}
